server/src/main/java/com/flipkart/varadhi/web/AuthorizationHandlerBuilder.java

```diff
@@ -35,24 +33,25 @@ public AuthorizationHandlerBuilder(List<String> superUsers, AuthorizationProvide
         this.provider = Objects.requireNonNull(provider, "Authorization Provider is null");
     }
 
-    public AuthorizationHandler build(PermissionAuthorization requiredAuthorization) {
-        return new AuthorizationHandler(requiredAuthorization);
+    public AuthorizationHandler build(ResourceAction authorizationOnResource) {
+        return new AuthorizationHandler(authorizationOnResource);
```

Comment:
since resource hierarchy is providing complete path, PermissionsAuthorization is not needed and replaced with action only